ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -191,13 +208,46 @@ private static SourceDescriptionWithWarnings describeSource(
       ), statementText);
     }
 
-    Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescription = Optional.empty();
+    final List<RunningQuery> sourceQueries = getQueries(ksqlEngine,
+        q -> q.getSourceNames().contains(dataSource.getName()));
+    final List<RunningQuery> sinkQueries = getQueries(ksqlEngine,
+        q -> q.getSinkName().equals(dataSource.getName()));
+
+    Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescriptionOptional = Optional
+        .empty();
+    final List<SourceConsumerGroupOffsets> sourceConsumerOffsets = new ArrayList<>();
     final List<KsqlWarning> warnings = new LinkedList<>();
     if (extended) {
       try {
-        topicDescription = Optional.of(
-            serviceContext.getTopicClient().describeTopic(dataSource.getKafkaTopicName())
-        );
+        final String kafkaTopicName = dataSource.getKafkaTopicName();
+        final TopicDescription topicDescription = serviceContext.getTopicClient()
+                .describeTopic(kafkaTopicName);
+        topicDescriptionOptional = Optional.of(topicDescription);
+        for (RunningQuery sourceQuery : sourceQueries) {
+          final QueryId queryId = sourceQuery.getId();
+          final String persistenceQueryPrefix =
+              ksqlConfig.getString(KsqlConfig.KSQL_PERSISTENT_QUERY_NAME_PREFIX_CONFIG);
+          final String applicationId = getQueryApplicationId(
+              KsqlConfig.getServiceId(ksqlConfig),
+              persistenceQueryPrefix,
+              queryId
+          );
+          final Map<TopicPartition, OffsetAndMetadata> topicAndConsumerOffsets =
+              serviceContext.getConsumerGroupClient().listConsumerGroupOffsets(applicationId);
+          final Map<TopicPartition, ListOffsetsResultInfo> topicAndStartOffsets =
+              serviceContext.getTopicClient().listTopicStartOffsets(kafkaTopicName);
+          final Map<TopicPartition, ListOffsetsResultInfo> topicAndEndOffsets =
+              serviceContext.getTopicClient().listTopicEndOffsets(kafkaTopicName);
```

Comment:
I think this is a bug: this is getting the offsets of the data source's sink topic, i.e. `kafkaTopicName` and comparing this to the consumer group offsets of queries writing into the sink topic.  Those queries won't be consuming `kafkaTopicName`, they'll be producing to it.  

Or am I missing somethinng?